test/adaptive_load/fake_plugins/fake_step_controller/fake_step_controller_test.cc

```diff
@@ -0,0 +1,208 @@
+            kExpectedValue);
+}
+
+TEST(FakeStepController, IsConvergedInitiallyReturnsFalse) {
+  FakeStepController step_controller(FakeStepControllerConfig{}, CommandLineOptions{});
+  EXPECT_FALSE(step_controller.IsConverged());
+}
+
+TEST(FakeStepController, IsConvergedReturnsFalseAfterBenchmarkResultWithoutPositiveScore) {
+  FakeStepController step_controller(FakeStepControllerConfig{}, CommandLineOptions{});
+  BenchmarkResult benchmark_result;
+  step_controller.UpdateAndRecompute(benchmark_result);
+  EXPECT_FALSE(step_controller.IsConverged());
+}
+
+TEST(FakeStepController, IsConvergedReturnsTrueAfterBenchmarkResultWithPositiveScore) {
+  FakeStepController step_controller(FakeStepControllerConfig{}, CommandLineOptions{});
+  BenchmarkResult benchmark_result;
+  MetricEvaluation* evaluation = benchmark_result.mutable_metric_evaluations()->Add();
+  evaluation->set_threshold_score(1.0);
+  step_controller.UpdateAndRecompute(benchmark_result);
+  EXPECT_TRUE(step_controller.IsConverged());
+}
+
+TEST(FakeStepController, IsDoomedReturnsFalseAfterSuccessfulBenchmarkResult) {
+  FakeStepController step_controller(FakeStepControllerConfig{}, CommandLineOptions{});
+  BenchmarkResult benchmark_result;
+  benchmark_result.mutable_status()->set_code(::grpc::OK);
+  step_controller.UpdateAndRecompute(benchmark_result);
+  std::string doomed_reason;
+  EXPECT_FALSE(step_controller.IsDoomed(doomed_reason));
+}
+
+TEST(FakeStepController, IsDoomedDoesNotWriteDoomedReasonAfterSuccessfulBenchmarkResult) {
+  FakeStepController step_controller(FakeStepControllerConfig{}, CommandLineOptions{});
+  BenchmarkResult benchmark_result;
+  benchmark_result.mutable_status()->set_code(::grpc::OK);
+  step_controller.UpdateAndRecompute(benchmark_result);
+  std::string doomed_reason = "untouched";
+  ASSERT_FALSE(step_controller.IsDoomed(doomed_reason));
+  EXPECT_EQ(doomed_reason, "untouched");
+}
+
+TEST(FakeStepController, IsDoomedReturnsTrueAfterFailedBenchmarkResult) {
+  FakeStepController step_controller(FakeStepControllerConfig{}, CommandLineOptions{});
+  BenchmarkResult benchmark_result;
+  benchmark_result.mutable_status()->set_code(::grpc::INTERNAL);
+  benchmark_result.mutable_status()->set_message("error from nighthawk");
+  step_controller.UpdateAndRecompute(benchmark_result);
+  std::string doomed_reason;
+  EXPECT_TRUE(step_controller.IsDoomed(doomed_reason));
+}
+
+TEST(FakeStepController, IsDoomedSetsDoomedReasonToStatusMessageAfterFailedBenchmarkResult) {
+  const std::string kErrorMessage = "error from nighthawk";
+  FakeStepController step_controller(FakeStepControllerConfig{}, CommandLineOptions{});
+  BenchmarkResult benchmark_result;
+  benchmark_result.mutable_status()->set_code(::grpc::INTERNAL);
+  benchmark_result.mutable_status()->set_message(kErrorMessage);
+  step_controller.UpdateAndRecompute(benchmark_result);
+  std::string doomed_reason;
+  ASSERT_TRUE(step_controller.IsDoomed(doomed_reason));
```

Comment:
Why ASSERT_TRUE rather than EXPECT_TRUE here?  Won't the code after still work?

Also, I honestly think that this and the previous test are essentially testing two very similar parts of the same thing, and we could combine them.  But that's an opinion, so I'll leave it up to you.

Reply:
Changed to EXPECT_TRUE.

Deleted the previous test that was a subset of this one.